ocaml/xapi/storage_access.ml

```diff
@@ -1007,12 +989,15 @@ let on_xapi_start ~__context =
         >>| fun t ->
         Client.list ~t ~prefix:!Storage_interface.queue_name ~filter:`Alive ()
         >>| fun running_smapiv2_driver_queues ->
-        List.filter
-          (fun driver ->
-             List.exists
-               (Xstringext.String.endswith driver)
-               running_smapiv2_driver_queues)
-          smapiv2_drivers
+        running_smapiv2_driver_queues
+        (* The results include the prefix itself, but that is the main storage
+           queue, we don't need it *)
+        |> List.filter ((<>) !Storage_interface.queue_name)
+        |> List.map (fun driver ->
+            (* Get the last component of the queue name: org.xen.xapi.storage.sr_type -> sr_type *)
+            (* split_on_char returns a non-empty list *)
+            String.split_on_char '.' driver |> List.rev |> List.hd
```

Comment:
This is required because we want to register unknown drivers from SMAPIv3, right?

Reply:
Yep, exactly.